### DISCUSSION TITLE
emit with array from blade

### DISCUSSION THREAD

**Discussion by animena** [livewire/livewire#4623] [1. Help] [answered]
Hello all, I have an issue with emitting an array of data from blade, the problem started with finding the best way for exchanging data between components and I found the only solution is to emit data from one component to another (correct me if I'm wrong) while I can emit array from PHP, I can't from blade at least without encoding it to json? the idea is, I don't want to create a function in every class to emit data to an event, but just emitting it directly from the blade and having the flexibility of working with the data from the array from the other components that listening to the event. or, if there is a better way to send data from all the components that loaded to the page, with maybe a global variable, that will also work in my case, which I tried AlpineJs but I run into a limit with using it.

Here is an example of the code:
```

@foreach($usedfields as $Id=>$field)
<div class="cursor-pointer" wire:click="$emit('show-side-bar', $usedfields)">
click here
</div>
@endforeach


the $usedfields


$this->usedfields     = (!empty($this->draft->data["usedfields"]))? collect($this->draft->data["usedfields"]) : collect([]);


from component listening to the event


protected $listeners = ['show-side-bar' => 'show'];

    public function show($data)
    {
            $this->sidebar = $data;
    }

```
the $usedfields is the array that I want to emit to as "show-side-bar" event, and from another component to get it as an array, I'm able to send a value for example, but not an array of data.

Thanks a lot for your help.

**Comment by elfeffe** [comment, marked as answer]
you have to jsonencode the array I think

**Comment by joshhanley** [comment]
@animena yeah @elfeffe is correct, due to the data being echo'd in the html (rather than serialised as part of the request when done from php), you will need to json encode it to make it safe.

To do this, make use of the new `@js` helper like `wire:click="$emit('show-side-bar', @js($usedfields))"`. See here in the docs https://laravel-livewire.com/docs/2.x/alpine-js#js-directive

Hope this helps!
